File: python/agregar_producto.py
from flask import Flask, request, jsonify
import mysql.connector

app = Flask(__name__)

# Configuración de la base de datos
db_config = {
    'host': 'localhost',
    'user': 'root',
    'password': 'root',
    'database': 'bd_unificada'
}

@app.route('/agregar_producto', methods=['POST'])
def agregar_producto():
    try:
        # Recibir datos JSON del cuerpo de la solicitud
        data = request.get_json()

        # Extraer campos
        nombre = data.get('nombre')
        tipo = data.get('tipo')
        talla = data.get('talla')
        color = data.get('color')
        cantidad_stock = data.get('cantidad_stock')
        precio = data.get('precio')
        id_ubicacion = data.get('id_ubicacion')

        # Validación básica
        if not all([nombre, tipo, talla, color, cantidad_stock, precio, id_ubicacion]):
            return jsonify({'error': 'Todos los campos son requeridos'}), 400

        # Conexión a la base de datos
        conn = mysql.connector.connect(**db_config)
        cursor = conn.cursor()

        # Consulta de inserción
        insert_query = """
            INSERT INTO producto (nombre, tipo, talla, color, cantidad_stock, precio, id_ubicacion)
            VALUES (%s, %s, %s, %s, %s, %s, %s)
        """
        cursor.execute(insert_query, (nombre, tipo, talla, color, cantidad_stock, precio, id_ubicacion))
        conn.commit()

        # Cierre de conexión
        cursor.close()
        conn.close()

        return jsonify({'mensaje': 'Producto agregado correctamente'}), 201

    except mysql.connector.Error as err:
        return jsonify({'error': str(err)}), 500

# Punto de entrada del microservicio
if __name__ == '__main__':
    app.run(debug=True, port=5001)

File: node/agregar_producto.js
const express = require('express');
const mysql = require('mysql2');
const app = express();

// Middleware para leer JSON
app.use(express.json());

// Configuración de la base de datos
const db = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'root',
  database: 'bd_unificada'
});

// Conexión a la base de datos
db.connect((err) => {
  if (err) {
    console.error('Error al conectar a la base de datos:', err.message);
    process.exit(1);
  }
  console.log('Conectado a la base de datos MySQL');
});

// Ruta POST para agregar producto
app.post('/agregar_producto', (req, res) => {
  const {
    nombre,
    tipo,
    talla,
    color,
    cantidad_stock,
    precio,
    id_ubicacion
  } = req.body;

  // Validación básica
  if (!nombre || !tipo || !talla || !color || !cantidad_stock || !precio || !id_ubicacion) {
    return res.status(400).json({ error: 'Todos los campos son requeridos' });
  }

  const insertQuery = `
    INSERT INTO producto (nombre, tipo, talla, color, cantidad_stock, precio, id_ubicacion)
    VALUES (?, ?, ?, ?, ?, ?, ?)
  `;

  const values = [nombre, tipo, talla, color, cantidad_stock, precio, id_ubicacion];

  db.query(insertQuery, values, (err, results) => {
    if (err) {
      console.error('Error al insertar producto:', err.message);
      return res.status(500).json({ error: err.message });
    }
    res.status(201).json({ mensaje: 'Producto agregado correctamente' });
  });
});

// Iniciar servidor
app.listen(5001, () => {
  console.log('Microservicio "agregar_producto" escuchando en http://localhost:5001');
});
